src/ppom-controller.ts

```diff
@@ -470,6 +494,20 @@ export class PPOMController extends BaseControllerV2<
     this.update((draftState) => {
       draftState.chainStatus = chainStatus;
     });
+  }
+
+  /*
+   * The function adds new network to chainStatus list.
+   */
+  #onNetworkChange(networkControllerState: NetworkState): void {
+    this.#chainId = addHexPrefix(networkControllerState.providerConfig.chainId);
+    this.#updateChainStatus(this.#chainId);
+    Object.values(networkControllerState.networkConfigurations).forEach(
+      (networkConfig) => {
+        const chainId = addHexPrefix(networkConfig.chainId);
+        this.#updateChainStatus(chainId);
+      },
+    );
```

Comment:
We want to cache files only for recently visited networks (visited in past 1 week)

Reply:
what exactly does "visited" mean here? sounds like you don't consider a network "visited" just because it shows up in the list of added networks. Does that mean "visited" is only true for the selected chain and any chains used inside `usePPOM()`?

Reply:
`lastVisited` is the time network was last visited. We fetch files only for networks visited in last 1 week. If we fetch files for all possible networks it will be too many files.

Reply:
If we can know when user connects to a new network we can add it to `chainStatus` it will be useful as it will also give chance to download network files before transaction comes.